src/book.py

```diff
@@ -179,6 +180,77 @@ def _read_coinbase(self, file_path: Path) -> None:
+        #   dup_state["deposit"] == 1: Deposit is broadcast to blockchain            <-- Taxable event (is in public trade history)
+        #   dup_state["deposit"] == 0: Deposit is credited to Kraken account         <-- Skipped
+        #   dup_state["withdrawal"] == 1: Withdrawal is requested in Kraken account  <-- Skipped
+        #   dup_state["withdrawal"] == 0: Withdrawal is broadcast to blockchain      <-- Taxable event (is in public trade history)
+        dup_state, dup_skip = {"deposit": 0, "withdrawal": 0}, {"deposit": 0, "withdrawal": 1}
+        # See: https://support.kraken.com/hc/en-us/articles/360001169443-Why-there-are-duplicate-entries-for-deposits-withdrawals
+
+        with open(file_path, encoding="utf8") as f:
+            reader = csv.reader(f)
+
+            # Skip header.
+            next(reader)
+
+            for txid, refid, _utc_time, _type, subtype, aclass, _asset, _amount, _fee, balance in reader:
+                row = reader.line_num
+
+                # Skip "duplicate entries" for deposits / withdrawals
+                if _type in dup_state.keys():
+                    dup_state[_type] = (dup_state[_type] + 1) % 2
+                    if dup_state[_type] == dup_skip[_type]:
+                        continue
+
+                # Parse data.
+                utc_time = datetime.datetime.strptime(_utc_time, "%Y-%m-%d %H:%M:%S").replace(tzinfo=datetime.timezone.utc)
+                change = float(_amount)
+                coin = kraken_asset_map.get(_asset, _asset)
+                fee = float(_fee)
+                operation = operation_mapping.get(_type)
+                if operation is None:
+                    if _type == "trade":
+                        operation = "Sell" if change < 0 else "Buy"
+                    elif _type in ["margin trade", "rollover", "settled"]:
+                        log.error(f"{file_path}: {row}: Margin trading is currently not supported. Please create an Issue or PR.")
+                        raise RuntimeError
+                    else:
+                        log.error(f"{file_path}: {row}: Other order type '{_type}' is currently not supported. Please create an Issue or PR.")
+                        raise RuntimeError
+                change = abs(change)
+
+                # Validate data.
+                assert operation
+                assert coin
+                assert change
+
+                self.append_operation(operation, utc_time, platform,
+                                      change, coin, row, file_path)
+
+                if fee != 0:
+                    self.append_operation("Fee", utc_time, platform,
+                                          fee, coin, row, file_path)
+
+        assert dup_state["deposit"] == 0, "Orphaned deposit. (Must always come in pairs). Is your file corrupted?"
+        assert dup_state["withdrawal"] == 0, "Orphaned withdrawal. (Must always come in pairs). Is your file corrupted?"
```

Comment:
If the change the calculation/if-case order, we have to adjust the assert statment to `==1`

Reply:
I don't think so! If deposits come in pairs, the counter will always be zero. Same for the withdrawals.

Reply:
You are right.

Reply:
I thought you switch the if-case with the calculation. Like so. Less code. Does the same. :D

```
                    if dup_state[_type] == dup_skip[_type]:
                        continue
                    dup_state[_type] = (dup_state[_type] + 1) % 2
```

Reply:
But my thought is stupid XD

Reply:
No problem! :D